examples/cookbook/recipe_three_pool_advice.cpp

```diff
@@ -0,0 +1,47 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory
+//
+// Created by David Beckingsale, david@llnl.gov
+// LLNL-CODE-747640
+//
+// All rights reserved.
+//
+// This file is part of Umpire.
+//
+// For details, see https://github.com/LLNL/Umpire
+// Please also see the LICENSE file for MIT license.
+//////////////////////////////////////////////////////////////////////////////
+#include "umpire/strategy/DynamicPool.hpp"
+#include "umpire/strategy/AllocationAdvisor.hpp"
+
+#include "umpire/Allocator.hpp"
+#include "umpire/ResourceManager.hpp"
+
+#include "umpire/util/Macros.hpp"
+
+#include <iostream>
+
+int main(int, char**) {
+  auto& rm = umpire::ResourceManager::getInstance();
+
+  auto allocator = rm.getAllocator("UM");
+
+  /*
+   * Create an allocator that applied "PREFFERED_LOCATION" advice to set the
+   * GPU as the preferred location.
+   */
+  auto preferred_location_allocator =
+    rm.makeAllocator<umpire::strategy::AllocationAdvisor>(
+      "preferred_location_host", allocator, "PREFERRED_LOCATION");
```

Comment:
How do programmers know which string to use for advice?  Are they listed someplace besides source code?

Reply:
Good point - only in the code I think. I create a ticket to document this better.